R/model_performance.R

```diff
@@ -139,3 +147,57 @@ model_performance_accuracy <- function(tp, fp, tn, fn) {
   (tp + tn)/(tp + fp + tn + fn)
 }
 
+model_performance_macro_f1 <- function(predicted, observed){
+  predicted_vectorized <- turn_porbs_into_vector(predicted)
+  confusion_matrixes <- calculate_confusion_matrixes(predicted_vectorized, observed)
+  f1_scores <- sapply(confusion_matrixes, function(x){
+    model_performance_f1(x$tp, x$fp, x$tn, x$fn)
+  })
+  mean(f1_scores)
+}
+
+model_performance_micro_f1 <- function(predicted, observed){
+  # For case where each point can be assigned only to one class micro_f1 equals acc
+  model_performance_accuracy_multi(predicted, observed)
+}
+
+model_performance_weighted_macro_f1 <- function(predicted, observed){
+  predicted_vectorized <- turn_porbs_into_vector(predicted)
+  confusion_matrixes <- calculate_confusion_matrixes(predicted_vectorized, observed)
+  f1_scores <- sapply(confusion_matrixes, function(x){
+    model_performance_f1(x$tp, x$fp, x$tn, x$fn)
+  })
+  weighted.mean(f1_scores, prop.table(table(observed))[names(confusion_matrixes)])
+}
+
+model_performance_accuracy_multi <- function(predicted, observed){
+  predicted_vectorized <- turn_porbs_into_vector(predicted)
+  mean(predicted_vectorized == observed)
+}
+
+model_performance_weighted_macro_auc <- function(predicted, observed) {
+  observed <- as.character(observed)
+  auc_scores <- sapply(unique(observed), function(x){
+    model_performance_auc(predicted[,x], as.numeric(observed == x))
+  })
+  weighted.mean(auc_scores, prop.table(table(observed))[unique(observed)])
+}
+
+turn_porbs_into_vector <- function(observed) {
+  apply(observed, 1, function(x){
+    colnames(observed)[which.max(x)]
+  })
+}
+
+calculate_confusion_matrixes <- function(predicted, observed){
+  observed <- as.character(observed)
```

Comment:
what would happen is one of these conditions are not satisfied (eg all predictions are TRUE)
would it work or cause errors?

Reply:
If added ifelse statement that checks if one of the values is NaN before assigning it to a list